pkg/ui/react-app/src/pages/graph/GraphControls.tsx

```diff
@@ -142,6 +144,19 @@ class GraphControls extends Component<GraphControlsProps> {
             <FontAwesomeIcon icon={faChartArea} fixedWidth />
           </Button>
         </ButtonGroup>
+
+        <Input
+          type="select"
+          value={this.props.maxSourceResolution}
+          onChange={e => this.props.onChangeMaxSourceResolution(e.target.value)}
```

Comment:
in general we try to avoid binding callbacks at render time with the arrow function in order to not trigger a diff. Do you think this is a small enough component that we can afford a diff? Or should we pre-bind?

Reply:
I was not aware of this. By pre-binding do you mean something like this?
```js
constructor(props) {
  super(props);

  this.handleMaxSourceResChange = this.handleMaxSourceResChange.bind(this);
}

handleMaxSourceResChange(event) {
  this.props.onChangeMaxSourceResolution(event.target.value)
}

render(
  <Input onChange={this.handleMaxSourceResChange} />
)
```
Sorry for the silly questions, I don't have much experience with class-based components.